helix-core/src/match_brackets.rs

```diff
@@ -161,18 +207,72 @@ fn surrounding_bytes(doc: &Rope, node: &Node) -> Option<(usize, usize)> {
     Some((start_byte, end_byte))
 }
 
+/// Checks if `node` or its siblings (at most MATCH_LIMIT nodes) is the specified openiing char
+///
+/// # Returns
+///
+/// The position of the found node or `None` otherwise
+fn find_open_pair(doc: RopeSlice, node: Option<Node>, open: char) -> Option<usize> {
+    iter::successors(node, |node| node.prev_sibling())
+        .take(MATCH_LIMIT)
+        .find_map(|node| {
+            let (pos, c) = as_char(doc, &node)?;
+            (c == open).then_some(pos)
+        })
+}
+
+/// Tests if this node is a pair close char and returns the expected open char
+fn as_close_pair(doc: RopeSlice, node: &Node) -> Option<char> {
+    let close = as_char(doc, node)?.1;
+    PAIRS
+        .iter()
+        .find_map(|&(open, close_)| (close_ == close).then_some(open))
+}
+
+/// Checks if `node` or its siblings (at most MATCH_LIMIT nodes) is the specified closing char
+///
+/// # Returns
+///
+/// The position of the found node or `None` otherwise
+fn find_close_pair(doc: RopeSlice, node: Option<Node>, close: char) -> Option<usize> {
+    iter::successors(node, |node| node.next_sibling())
+        .take(MATCH_LIMIT)
+        .find_map(|node| {
+            let (pos, c) = as_char(doc, &node)?;
+            (c == close).then_some(pos)
+        })
+}
+
+/// Tests if this node is a pair close char and returns the expected open char
+fn as_open_pair(doc: RopeSlice, node: &Node) -> Option<char> {
+    let close = as_char(doc, node)?.1;
+    PAIRS
+        .iter()
+        .find_map(|&(open, close_)| (close_ == close).then_some(open))
+}
```

Comment:
I think you have implemented the exact same logic twice, but just gave it a different name. Was this on purpose? If not I would suggest finding a better name, and re-use the same implementation

Reply:
ah right I copied that function and then wanted to change it but forgot. I pushed a fix

Reply:
Ok I don't see this issue anymore with recent push :)